PerfectDay/Repositories/IRepositoryUser.cs

```diff
@@ -0,0 +1,43 @@
+﻿using Microsoft.EntityFrameworkCore;
+using PerfectDay.Entities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace PerfectDay.Repositories
+{
+    public class IRepositoryUser<TEntity> : IRepository<TEntity> where TEntity : class
+    {
+        ApplicationContex _context;
+        DbSet<TEntity> _dbSet;
```

Comment:
Please use more appropriate name. We already know that we change only users here.